src/Logging/Logging.AzureAppServices/src/BatchingLoggerProvider.cs

```diff
@@ -143,10 +143,7 @@ private void Start()
                 new BlockingCollection<LogMessage>(new ConcurrentQueue<LogMessage>(), _queueSize.Value);
 
             _cancellationTokenSource = new CancellationTokenSource();
-            _outputTask = Task.Factory.StartNew<Task>(
-                ProcessLogQueue,
-                null,
-                TaskCreationOptions.LongRunning);
+            _outputTask = Task.Run(ProcessLogQueue);
```

Comment:
Do you need to `Task.Run`?  Calling `ProcessLogQueue` should be enough.

Reply:
As I understand it, Calling `ProcessLogQueue()` directly would start running `ProcessLogQueue` inline, on the same thread, instead of using a `ThreadPool` thread. It would only return and complete the constructor after the first `await` is hit (either `WriteMessagesAsync` if that completes asynchronously, or `IntervalAsync`).

While that's _probably_ fine, it seems undesirable to be doing real work here rather than just kicking off the background work, but I'm happy to defer to you on that 🙂